### PR TITLE
download s3 object concurrently

### PR DESCRIPTION
**Pull request by guangrui-cloudnatix**
to be tested e2e

### CODE REVIEW

File: engine/internal/huggingface/download_test.go [hunk 1]
```diff
@@ -64,7 +65,7 @@ func TestDownloadModelFiles(t *testing.T) {
 				_ = os.RemoveAll(dir)
 			}()
 
-			err = DownloadModelFiles(tc.s3Client, "/src", dir)
+			err = DownloadModelFiles(context.TODO(), tc.s3Client, "/src", dir)
```
Comment (kkaneda) [inline engine/internal/huggingface/download_test.go:68]:
```suggestion
			err = DownloadModelFiles(context.Background(), tc.s3Client, "/src", dir)
```

File: engine/internal/s3/client.go [hunk 2]
```diff
@@ -1,42 +1,50 @@
...
 
-// Download downloads an object from S3.
-func (c *Client) Download(w io.WriterAt, key string) error {
-	downloader := s3manager.NewDownloaderWithClient(c.svc)
-	_, err := downloader.Download(w, &s3.GetObjectInput{
+// Download uses a download manager to download an object from a bucket.
+// The download manager gets the data in parts and writes them to a buffer until all of
+// the data has been downloaded.
+func (c *Client) Download(ctx context.Context, w io.WriterAt, key string) error {
+	var partMiBs int64 = 10
```
Comment (kkaneda) [inline engine/internal/s3/client.go:43]:
```suggestion
	const partMiBs int64 = 10
```

Also the part size can be bigger? Not sure though.

**Review by kkaneda — APPROVED** [review]
LG!

File: engine/internal/s3/client.go (hunk 2 above)
Comment (guangrui-cloudnatix) [reply engine/internal/s3/client.go:43]: Can be larger. It basically depends on if we want concurrent downloading for a small or medium object, e.g. 50MB or 100MB object. Any suggestions on what default value to use?

Comment (kkaneda) [reply engine/internal/s3/client.go:43]: So 

- `model.gguf` is the largest as it contains the entire model. Can be >10GiB
- safetensors files is severeal GiB.
- Other json files are few KB or MB. 

Maybe use 128MB as a chunk size that gives sufficient concurrency for large file download.

**Pull request merged**